### PR TITLE
Include parameter to indicate endpoint

### PR DESCRIPTION
**Pull request by sarroutbi**
Resolves: #116

### CODE REVIEW

**Comment by sarroutbi** [comment]
WIP. Unit Tests and man pages pending.

**Comment by sarroutbi** [comment]
Blocked until #119 is merged

**Review by sergio-correia — CHANGES_REQUESTED** [review]
This looks good overall, but please fix the indent to match the file you are changing: e.g.: it seems you are using 2 spaces in some places like `socket.h` and `tangd.c` (perhaps others as well)

File: src/tangd.c
Comment (sergio-correia) [inline src/tangd.c:194]: Instead of `NULL`, why don't you leave as it was here, i.e., the default (with no `endpoint`), and then you adjust it in the case there is an endpoint specified?

File: src/tangd.c
Comment (sergio-correia) [inline src/tangd.c:204]: A better name for these variables would be `adv_` and `adv_thp_`, instead of `adv2`, which does not mean much.

File: src/tangd.c
Comment (sergio-correia) [inline src/tangd.c:304]: Instead of passing `endpoint` here and  later on in `run_service()`, it would probably be better to adjust `s_dispatch` for the endpoint,  when required.

**Review by sergio-correia — CHANGES_REQUESTED** [review]

File: src/tangd.c
Comment (sarroutbi) [reply src/tangd.c:304]: Not sure what you mean with this. `run_service()` calls `spawn_process()` and `spawn_process()` calls `process_request()`. This last function adapts the s_dispatch with the endpoint, so it needs user provided endpoint to adjust it

**Review by sergio-correia — APPROVED** [review]
LGTM, thanks for this work!

**Pull request merged**
